### DISCUSSION TITLE
Roadrunner for non-PHP apps?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [roadrunner-server/roadrunner#1319] [Q&A] [answered]
Really simple, really basic.

Can I use roadrunner for a mixed application stack like Node or Golang?

Thanks!

**Comment by rustatian** [comment, marked as answer]
Hey @IngwiePhoenix 👋🏻 
Sure, RR itself doesn't know anything about PHP running inside the process. You'll only need to implement the `goridge` protocol in JS.
Here you may have a look at the Golang part (https://github.com/roadrunner-server/goridge)
PHP part: https://github.com/spiral/goridge-php
And specs: https://github.com/roadrunner-server/goridge/blob/master/pkg/frame/frame.md.
